engine/gameobject/src/gameobject/gameobject.cpp

```diff
@@ -1038,6 +1046,12 @@ namespace dmGameObject
         return SetIdentifier(hcollection->m_Collection, instance, id);
     }
 
+    Result SetConstructedIdentifier(HCollection hcollection, HInstance instance)
+    {
+        instance->m_Generated = 1;
+        return SetIdentifier(hcollection->m_Collection, instance, ConstructInstanceId());
+    }
```

Comment:
I don't think we need this function?
The reason is I don't think we need the `m_Generated`?
The only place it's used is for the debug functionality of checking to call the `dmHashReverseErase64(instance->m_Identifier);` and `dmScript::ReleaseHash(L, instance->m_Identifier);`

Both of those functions can (should at least!) handle non existent entries.
So, we can reduce complexity altogether by removing the m_Generated flag.

Reply:
I think the biggest issue here not non-existent entries, but removing some hashes which are in use.
Let me explain.
We don't have reference counters for hashes.
User may have just a "/go" in two collections, and it's not factory generated one, just a collection. As a result, when such `/go` removed in one place, it will be removed from the ReverseHashTable, and in all the places where `/go` used for the second `/go` it will be hard to debug.

That's why I mentioned that having a unique name to be able to freely release it from reverse table is important. Which is possible only for the cases we create them ourselves

Reply:
Sure I get it.
The problem though is that the id's you mention are thus "leaked", and will always stay in the reverse hash table 
I think that's not a "clean" solution either. Granted, we've had it for a long time, but I just don't like the fact that we can't treat all game objects or id's equal.

Also, I _really_ don't like that the api's gets so dictated by the debug functionality either.
E.g. this new function, giving us 2 functions to call, which is quite ambiguous.

I think a better way to solve this, is if the gameobject.cpp has it's own debug store of hashes, with ref counts.
And when they get deleted from there, we remove it from the actual reverse hash as well.

Reply:
I think I can remove this `SetConstructedIdentifier` function. Let me think about it a little bit